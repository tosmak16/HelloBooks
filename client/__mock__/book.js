const books = [
  {
    id: 1,
    bookId: 1,
    bookTitle: 'THE PEOPLE IN THE TREE',
    author: 'Y. Hayan',
    category: 'Adventure',
    isbn: '345798465',
    stocknumber: '2',
    image: 'l1.jpg',
    summary: 'Lorem Ipronic typesetting, remaininised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum',
  },
  {
    id: 2,
    bookId: 2,
    bookTitle: 'THE LATE PARADE',
    author: 'F. Adam',
    category: 'Adventure',
    isbn: '234560878',
    stocknumber: '0',
    image: 'l2.jpg',
    summary: 'It is a long established fact thaorem Ipsu uncover many web sites still in their infancy. Various versions have evolved over the years, sometimes by accident, sometimes on purpose (injected humour and the like)',
  },
  {
    id: 3,
    bookId: 3,
    bookTitle: 'I AM THE WOLF',
    author: 'L. Mark',
    category: 'Music',
    isbn: '4678980976',
    stocknumber: '5',
    image: 'l3.jpg',
    summary: 'There are many variations of passages of Lorem Ipsum available, but the majority orem Ipsum generators on the Internet tend to repeat predefined chunks as necessary',

  },
  {
    id: 4,
    bookId: 4,
    bookTitle: 'THE PALE NORTH',
    author: 'H. Clayton',
    category: 'Adventure',
    isbn: '34567887655',
    stocknumber: '3',
    image: 'l4.jpg',
    summary: 'The standard chunk of Lorem Ipsum used since the 1500s is reproduced below for those interested. Sections 1.10.32 and 1.10.33 from ',
  },
  {
    id: 5,
    bookId: 5,
    bookTitle: 'ADULTOLESENCE',
    author: 'G. Hannah',
    category: 'Romance',
    isbn: '45678974567',
    stocknumber: '10',
    image: 'l5.jpg',
    summary: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Urum.'
  },
  {
    id: 6,
    bookId: 6,
    bookTitle: 'WHAT IT MEANS WHEN A MAN FALL FROM SKY',
    author: 'L. Nneka',
    category: 'Adventure',
    isbn: '567244567',
    stocknumber: '10',
    image: 'l6.jpg',
    summary: 'Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium?'
  },
  {
    id: 7,
    bookId: 7,
    bookTitle: 'THE SONG AND THE SILENCE',
    author: 'Y. Johnson',
    category: 'Music',
    isbn: '64578974567',
    stocknumber: '10',
    image: 'l7.jpg',
    summary: 'But I must explain to you how all this mists no resultant pleasure  ducimus qui blanditiis praesentium voluptatum culpa qui officia ?'
  },
  {
    id: 8,
    bookId: 8,
    bookTitle: 'THE ILIAD HOMMER',
    author: 'P. Green',
    category: 'Action',
    isbn: '787890974567',
    stocknumber: '8',
    image: 'l8.jpg',
    summary: 'At vero eos et accusamus et iuse delectus, ut aut reiciendis voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repellat.'
  },
];

export default books;
