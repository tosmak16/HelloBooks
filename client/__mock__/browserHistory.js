import { createMemoryHistory } from 'react-router';


const browserHistory = createMemoryHistory('/');

export default browserHistory;
