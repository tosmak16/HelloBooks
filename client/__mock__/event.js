export const e = {
  target: {
    name: 'Tosmak',
    value: '10'
  },
};

export class event {
  static preventDefault() {

  }
}

