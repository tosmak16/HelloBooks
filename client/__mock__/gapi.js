const render = () => { };
const signin2 = {
  render,
};
global.gapi = {
  signin2
};
