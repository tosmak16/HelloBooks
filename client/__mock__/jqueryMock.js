import $ from 'jquery';
import React from 'react';

global.$ = global.jQuery = $;

$.prototype.modal = () => { };

global.Materialize = {
  toast: () => { }
};
