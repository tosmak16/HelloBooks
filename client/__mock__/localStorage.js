
export const localStorage = {
  setItem(key, value) {
    return Object.assign(window.localStorage, { [key]: value });
  },
  getItem(key) {
    return window.localStorage[key];
  },
  clear() {
    window.localStorage = {};
  },

  removeItem(key) {
    return delete window.localStorage[key];
  }

};
export default localStorage;
