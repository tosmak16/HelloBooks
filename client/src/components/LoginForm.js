import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Button, Icon, Input, Row } from 'react-materialize';

import userSignin from '../actions/loginAction';
import getbooks from '../actions/getBooks';
import log from '../../reducers/log';

class LoginForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: '',
      errors: ''
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  handleSubmit(e) {
    e.preventDefault();
    this.setState({ errors: '' });
    this.props.userSignin(this.state);
    this.props.getbooks(true);
  }

  render() {
    const { error } = this.props.log;
    return (
      <form onSubmit={ this.handleSubmit } className="form-signin" action="books.html" >
        <h4 className="form-signin-heading">Please log in</h4>
        {error && <p style={{ color: 'red' }} className="help-block">*{error}*</p>}
        <Row>
          <Input
            s={ 12 } label="Username"
            type="text" value={ this.state.username } name="username" id="inputUsernamelog" className="form-control validate"
            placeholder="Username"
            onChange={ this.handleChange }
            required
            autoFocus
          />
        </Row>

        <Row>
          <Input
            s={ 12 } label="Password"
            type="password"
            id="inputPassword"
            className="form-control validate"
            placeholder="Password" required
            autoFocus
            name="password"
            value={ this.state.password }
            onChange={ this.handleChange }
          />
        </Row>

        <div className="row">
          <div>
            <input
              defaultChecked="checked" name="group1" type="checkbox" value="" label="Remember me?"
              className="filled-in checkbox-orange" id="filled-in-box"
            />
            <label htmlFor="filled-in-box">Remember me?</label>
          </div>
        </div>
        <button id="loginbtn" className="btn btn-lg btn-primary  btn-block" type="submit">Sign in</button>
        <p>
          <label>
            <a href=""> Forgot Password ? </a>
          </label>
        </p>

        <a href="/signup" id="createbtn" className="" > <b>Create account</b></a>

      </form >
    );
  }
}

LoginForm.propTypes = {
  getbooks: PropTypes.func.isRequired,
  log: PropTypes.object.isRequired,
  userSignin: PropTypes.func.isRequired,

};

const mapStateToProps = state => ({
  log: state.log[0]
});

export default connect(mapStateToProps, { userSignin, getbooks })(LoginForm);
