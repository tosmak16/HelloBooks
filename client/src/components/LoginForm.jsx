import React from 'react';
import PropTypes from 'prop-types';
import { Input, Row } from 'react-materialize';
import $ from 'jquery';
import { ResetPasswordModal } from './modal/ResetPasswordModal';


/**
 * @export LoginForm
 * 
 * @description Login form un connected component
 * 
 * @class LoginForm
 * 
 * @extends {React.Component}
 */
export class LoginForm extends React.Component {
  /**
   * @description Creates an instance of LoginForm.
   * Login form  class method
   * 
   * @param {object} props
   * 
   * @memberof LoginForm
   */
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: '',
      errors: '',
      displayModal: false,
      userEmailText: ''
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.onSignIn = this.onSignIn.bind(this);
    this.handleGoogleSignin = this.handleGoogleSignin.bind(this);
    this.renderResetPasswordModal = this.renderResetPasswordModal.bind(this);
    this.handleClose = this.handleClose.bind(this);
    this.handleInputEmailChange = this.handleInputEmailChange.bind(this);
    this.handleSendMail = this.handleSendMail.bind(this);
  }
  /**
   * @memberof LoginForm
   * 
   * @returns {void}
   */
  componentDidMount() {
    $.getScript('https://apis.google.com/js/platform.js')
      .done(() => {
        // execute any gapi calls here...
        gapi.signin2.render('g-signin2', {
          scope: 'profile email',
          height: 40,
          width: 298,
          longtitle: false,
        });
      });
  }
  /**
   * @param {object} nextProps
   * 
   * @memberof LoginForm
   * 
   * @returns {void}
   */
  componentWillReceiveProps(nextProps) {
    const { resetPasswordStatus } = nextProps;
    if (resetPasswordStatus !== this.props.resetPasswordStatus) {
      this.setState({
        userEmailText: ''
      });
    }
  }
  /**
  * @param {object} response

  * @memberof LoginForm

  * @returns {void}
  */
  onSignIn(response) {
    this.props.googleAuthSignIn(response);
  }
  /**
   * @memberof LoginForm
   * @returns {void}
   */
  handleGoogleSignin() {
    $.getScript('https://apis.google.com/js/platform.js')
      .done(() => {
        // execute any gapi calls here...
        gapi.signin2.render('g-signin2', {
          scope: 'profile email',
          height: 40,
          width: 298,
          longtitle: false,
          onsuccess: this.onSignIn
        });
      });
  }

  /**
   * @param {event} event
   * 
   * @memberof LoginForm
   * 
   * @returns {void}
   */
  handleChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }
  /**
   * @param {object} event
   * 
   * @memberof LoginForm
   * 
   * @returns {void}
   */
  handleSubmit(event) {
    event.preventDefault();
    this.setState({ errors: '' });
    this.props.userSignin(this.state);
    this.props.getbooks(true);
  }
  /**
 * @param {object} event
 * 
 * @memberof LoginForm
 * 
 * @returns {void}
 */
  handleClose() {
    this.setState({
      displayModal: false
    });
  }

  /**
 * @param {object} event
 * 
 * @memberof LoginForm
 * 
 * @returns {void}
 */
  handleSendMail() {
    this.props.resetUserPassword(this.state.userEmailText);
  }

  /**
 * @param {object} event
 * 
 * @memberof LoginForm
 * 
 * @returns {void}
 */
  handleInputEmailChange(event) {
    this.setState({
      userEmailText: event.target.value
    });
  }
  /**
  * @param {object} event

  * @memberof LoginForm

  * @returns {void}
  */
  renderResetPasswordModal() {
    this.setState({
      displayModal: true
    });
  }
  /**
   * @returns {views} with input fields
   * 
   * @memberof LoginForm
   */
  render() {
    const { error } = this.props.login;
    return (
      <div>
        <form onSubmit={this.handleSubmit} className="form-signin" action="books.html" >
          <h5 className="form-signin-heading text-center pull-center">Please Login In</h5>
          {error && <p style={{ color: 'red' }} className="help-block">*{error}*</p>}
          <Row>
            <Input
              s={12}
              label="Username"
              type="text"
              value={this.state.username}
              name="username"
              id="inputUsernamelog"
              className="form-control validate"
              onChange={this.handleChange}
              required
              autoFocus
            />
          </Row>

          <Row>
            <Input
              s={12}
              label="Password"
              type="password"
              id="inputPassword"
              className="form-control validate"
              placeholder="Password"
              required
              autoFocus
              name="password"
              value={this.state.password}
              onChange={this.handleChange}
            />
          </Row>
          <div className="row">
            <div>
              <input
                defaultChecked="checked"
                name="group1"
                type="checkbox"
                value=""
                label="Remember me?"
                className="filled-in checkbox-orange"
                id="filled-in-box"
              />
              <label htmlFor="filled-in-box">Remember me?</label>
            </div>
          </div>
          <button
            id="loginbtn"
            className="btn btn-lg btn-primary  btn-block"
            type="submit"
          >Sign in</button>
          <div
            style={{
              marginLeft: '1px',
              marginTop: '10px',
            }}
            id="g-signin2"
            onClick={this.handleGoogleSignin}
            data-onsuccess={this.onSignIn}
          />
          <p>
            <label htmlFor="forgot_pass" >
              <button
                id="forgot_pass"
                type="button"
                onClick={this.renderResetPasswordModal}
              > Forgot Password ? </button>
            </label>
          </p>
          <a href="/signup" id="createbtn" className="" > <b>Create account</b></a>
        </form >
        {
          this.state.displayModal
          &&
          <ResetPasswordModal
            handleClose={this.handleClose}
            handleSendMail={this.handleSendMail}
            handleInputEmailChange={this.handleInputEmailChange}
            userEmailText={this.state.userEmailText}
          />
        }
      </div>
    );
  }
}
LoginForm.propTypes = {
  getbooks: PropTypes.func.isRequired,
  login: PropTypes.objectOf(PropTypes.any).isRequired,
  userSignin: PropTypes.func.isRequired,
  googleAuthSignIn: PropTypes.func.isRequired,
  resetUserPassword: PropTypes.func.isRequired,
  resetPasswordStatus: PropTypes.string.isRequired
};
export default LoginForm;
