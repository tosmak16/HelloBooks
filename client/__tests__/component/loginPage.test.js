import React from 'react';
import render from 'react-test-renderer';
import expect from 'expect';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { mount, shallow } from 'enzyme';


import ConnectedLoginPage, { Login } from '../../src/components/Login';
import { LoginForm } from '../../src/components/LoginForm';
import { e, event } from '../../__mock__/event';


const mockFuction = jest.fn();


describe('Login Page Component test', () => {
  const initialState = {
    login: [{
      isAuthenticated: false,
      data: {},
      error: '',
      isFetching: false,
      response: {}
    }],
    resetPassword: [{
      isSending: false,
      data: '',
      error: '',
    }]
  };
  const mockStore = configureMockStore();
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(initialState);

    wrapper = mount(<Provider store={store}><ConnectedLoginPage /></Provider>);
  });

  it('should check if LoginPage is defined', () => {
    expect(Login).toBeDefined();
  });

  it('should check if LoginForm is defined', () => {
    expect(LoginForm).toBeDefined();
  });

  it('should test and take snapshot of LoginForm', () => {
    const tree = render.create(<LoginForm resetPasswordStatus='' resetUserPassword={mockFuction} googleAuthSignIn={mockFuction} userSignin={mockFuction} getbooks={mockFuction} login={{ error: '' }} />);
    expect(tree).toMatchSnapshot();
  });

  it('should test and take snapshot of LoginPage', () => {
    const tree = render.create(<Login resetPasswordStatus='' resetUserPassword={mockFuction} googleAuthSignIn={mockFuction} userSignin={mockFuction} getbooks={mockFuction} login={{ error: 'error' }} />);
    expect(tree).toMatchSnapshot();
  });

  it('should test and take snapshot of connected loginpPage', () => {
    const tree = render.create(<connectLoginPage resetPasswordStatus='' resetUserPassword={mockFuction} googleAuthSignIn={mockFuction} userSignin={mockFuction} getbooks={mockFuction} log={{ error: 'error' }} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render the connected component', () => {
    expect(wrapper.find(ConnectedLoginPage).length).toEqual(1);
  });

  it('it should check Prop matches with initialState', () => {
    expect(wrapper.find(Login).prop('login')).toEqual(initialState.login[0]);
  });

  it('should test for handle change function', () => {
    const nextProps = {
      resetPasswordStatus: ''
    }
    wrapper = shallow(<LoginForm resetPasswordStatus='' resetUserPassword={mockFuction} googleAuthSignIn={mockFuction} userSignin={mockFuction} getbooks={mockFuction} login={{ error: 'error' }} />);
    wrapper.instance().handeleFuction = mockFuction;
    wrapper.update();
    wrapper.instance().handleChange(e);
    wrapper.update();
    wrapper.instance().componentWillReceiveProps(nextProps)
  });

  it('should test for handle submit function and error state', () => {
    const nextProps = {
      resetPasswordStatus: 'true'
    }
    wrapper = shallow(<LoginForm resetPasswordStatus='' resetUserPassword={mockFuction} googleAuthSignIn={mockFuction} userSignin={mockFuction} getbooks={mockFuction} login={{ error: 'error' }} />);
    wrapper.instance().handeleFuction = mockFuction;
    wrapper.update();
    wrapper.instance().handleSubmit(event);
    wrapper.update();
    wrapper.instance().componentWillReceiveProps(nextProps)
    wrapper.instance().onSignIn()
    wrapper.instance().handleGoogleSignin()
    wrapper.instance().handleClose()
    wrapper.instance().handleSendMail()
    wrapper.instance().handleInputEmailChange(e)
    wrapper.instance().renderResetPasswordModal()


  });
});
