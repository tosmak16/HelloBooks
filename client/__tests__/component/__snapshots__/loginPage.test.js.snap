// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page Component test should test and take snapshot of LoginForm 1`] = `
<div>
  <form
    action="books.html"
    className="form-signin"
    onSubmit={[Function]}
  >
    <h5
      className="form-signin-heading text-center pull-center"
    >
      Please Login In
    </h5>
    
    <div
      className="row"
    >
      <div
        className="col input-field s12"
      >
        <input
          autoFocus={true}
          className="form-control validate"
          defaultValue={undefined}
          id="inputUsernamelog"
          name="username"
          onChange={[Function]}
          placeholder={undefined}
          required={true}
          type="text"
          value=""
        />
        <label
          className=""
          data-error={undefined}
          data-success={undefined}
          htmlFor="inputUsernamelog"
        >
          Username
        </label>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col input-field s12"
      >
        <input
          autoFocus={true}
          className="form-control validate"
          defaultValue={undefined}
          id="inputPassword"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
        <label
          className=""
          data-error={undefined}
          data-success={undefined}
          htmlFor="inputPassword"
        >
          Password
        </label>
      </div>
    </div>
    <div
      className="row"
    >
      <div>
        <input
          className="filled-in checkbox-orange"
          defaultChecked="checked"
          id="filled-in-box"
          label="Remember me?"
          name="group1"
          type="checkbox"
          value=""
        />
        <label
          htmlFor="filled-in-box"
        >
          Remember me?
        </label>
      </div>
    </div>
    <button
      className="btn btn-lg btn-primary  btn-block"
      id="loginbtn"
      type="submit"
    >
      Sign in
    </button>
    <div
      data-onsuccess={[Function]}
      id="g-signin2"
      onClick={[Function]}
      style={
        Object {
          "marginLeft": "1px",
          "marginTop": "10px",
        }
      }
    />
    <p>
      <label
        htmlFor="forgot_pass"
      >
        <button
          id="forgot_pass"
          onClick={[Function]}
          type="button"
        >
           Forgot Password ? 
        </button>
      </label>
    </p>
    <a
      className=""
      href="/signup"
      id="createbtn"
    >
       
      <b>
        Create account
      </b>
    </a>
  </form>
</div>
`;

exports[`Login Page Component test should test and take snapshot of LoginPage 1`] = `
<div>
  <div
    className="row"
    id="logininForm"
  >
    <div>
      <form
        action="books.html"
        className="form-signin"
        onSubmit={[Function]}
      >
        <h5
          className="form-signin-heading text-center pull-center"
        >
          Please Login In
        </h5>
        <p
          className="help-block"
          style={
            Object {
              "color": "red",
            }
          }
        >
          *
          error
          *
        </p>
        <div
          className="row"
        >
          <div
            className="col input-field s12"
          >
            <input
              autoFocus={true}
              className="form-control validate"
              defaultValue={undefined}
              id="inputUsernamelog"
              name="username"
              onChange={[Function]}
              placeholder={undefined}
              required={true}
              type="text"
              value=""
            />
            <label
              className=""
              data-error={undefined}
              data-success={undefined}
              htmlFor="inputUsernamelog"
            >
              Username
            </label>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col input-field s12"
          >
            <input
              autoFocus={true}
              className="form-control validate"
              defaultValue={undefined}
              id="inputPassword"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              required={true}
              type="password"
              value=""
            />
            <label
              className=""
              data-error={undefined}
              data-success={undefined}
              htmlFor="inputPassword"
            >
              Password
            </label>
          </div>
        </div>
        <div
          className="row"
        >
          <div>
            <input
              className="filled-in checkbox-orange"
              defaultChecked="checked"
              id="filled-in-box"
              label="Remember me?"
              name="group1"
              type="checkbox"
              value=""
            />
            <label
              htmlFor="filled-in-box"
            >
              Remember me?
            </label>
          </div>
        </div>
        <button
          className="btn btn-lg btn-primary  btn-block"
          id="loginbtn"
          type="submit"
        >
          Sign in
        </button>
        <div
          data-onsuccess={[Function]}
          id="g-signin2"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": "1px",
              "marginTop": "10px",
            }
          }
        />
        <p>
          <label
            htmlFor="forgot_pass"
          >
            <button
              id="forgot_pass"
              onClick={[Function]}
              type="button"
            >
               Forgot Password ? 
            </button>
          </label>
        </p>
        <a
          className=""
          href="/signup"
          id="createbtn"
        >
           
          <b>
            Create account
          </b>
        </a>
      </form>
    </div>
  </div>
  <footer
    className="footer-distributed "
  >
    <div
      className="row"
    >
      <div
        className=" col l12 col m12 col s12"
      >
        <div
          className="footer-left col l4 col m4 col s12"
        >
          <h3>
            Hello
            <span>
              Books
            </span>
          </h3>
          <p
            className="footer-links"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Home
            </a>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Blog
            </a>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Pricing
            </a>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              About
            </a>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Faq
            </a>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Contact
            </a>
          </p>
          <p
            className="footer-company-name"
          >
            Tosmak © 2017
          </p>
        </div>
        <div
          className="footer-center col l4 col m4 col s12"
        >
          <div>
            <i
              className="fa fa-map-marker"
            />
            <p>
              <span>
                21 Ori-oke Street
              </span>
               Lagos, Nigeria
            </p>
          </div>
          <div>
            <i
              className="fa fa-phone"
            />
            <p>
              08127112053
            </p>
          </div>
          <div>
            <i
              className="fa fa-envelope"
            />
            <p>
              <a
                href="tosmak16@gmail.com"
                onClick={[Function]}
                style={Object {}}
              >
                tosmak16@gmail.com
              </a>
            </p>
          </div>
        </div>
        <div
          className="footer-right col l4 col m14 col s12"
        >
          <div
            className="footer-company-about"
          >
            <span>
              About Hello Book
            </span>
            <p>
              it is a simple application that helps manage a library and its processes like stocking,tracking and renting books
            </p>
          </div>
          <div
            className="footer-icons"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <i
                className="fa fa-facebook"
              />
            </a>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <i
                className="fa fa-twitter"
              />
            </a>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <i
                className="fa fa-linkedin"
              />
            </a>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <i
                className="fa fa-github"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;

exports[`Login Page Component test should test and take snapshot of connected loginpPage 1`] = `
<connectLoginPage
  getBooks={[Function]}
  googleAuthSignIn={[Function]}
  log={
    Object {
      "error": "error",
    }
  }
  resetPasswordStatus=""
  resetUserPassword={[Function]}
  userSignin={[Function]}
/>
`;
